public/controllers/agent/components/register-agent.js

```diff
@@ -221,8 +221,19 @@ export const RegisterAgent = withErrorBoundary(
     async getRemoteInfo() {
       try {
         const result = await WzRequest.apiReq('GET', '/agents/000/config/request/remote', {});
+        console.log(result, 'resultt+++')
+        console.log('remoteee+')
         const remote = ((result.data || {}).data || {}).remote || {};
-        return (remote[0] || {}).protocol !== 'tcp' && (remote[0] || {}).protocol[0] !== 'TCP';
+        const remoteFiltered = remote.filter((item, key) => {
+          if (item.connection === 'secure') {
+            if(item.protocol.length === 1 && item.protocol[0] == 'UDP') {
```

Comment:
**nitpick**: The `protocol` property is an array that can contain `'TCP'`, `'UDP'` or both. On this line, we are trusting the `'UDP'` value will be at the 0 index. From my tests, configurating the manager as:
```xml
<remote>
    <connection>secure</connection>
    <port>1514</port>
    <protocol>UDP,tcp</protocol>
    <queue_size>131072</queue_size>
  </remote>
```

The API response to `/agents/000/config/request/remote`:
```json
{
  "data": {
    "remote": [
      {
        "connection": "secure",
        "ipv6": "no",
        "protocol": [
          "TCP",
          "UDP"
        ],
        "port": "1514",
        "queue_size": "131072"
      }
    ]
  },
  "error": 0
}
```

Note the order is `'TCP'` and `'UDP'` that is different of the configuration. This means the API response sorts the protocols. I would avoid checking the 0 index to know what protocol to use and see another way. The current solution could cause bugs if the API response is not sorted as it does at this moment.

References: https://github.com/wazuh/wazuh-kibana-app/issues/4318#issuecomment-1261081676